tests/try_into.rs

```diff
@@ -0,0 +1,89 @@
+#![allow(dead_code)]
+
+// TryInto is still an unstable trait. In the meanwhile, use a trick
+// to define the trait (as well as TryFrom) without relying on
+// #![feature(try_from)].
+#![no_std]
+extern crate std as ruststd;
+mod std {
+    pub mod convert {
+        pub trait TryFrom<T>: Sized {
+            type Error;
+            fn try_from(value: T) -> Result<Self, Self::Error>;
+        }
+
+        pub trait TryInto<T>: Sized {
+            type Error;
+            fn try_into(self) -> Result<T, Self::Error>;
+        }
+
+        impl<T, U> TryInto<U> for T where U: TryFrom<T>
+        {
+            type Error = U::Error;
+
+            fn try_into(self) -> Result<U, U::Error> {
+                U::try_from(self)
+            }
+        }
+    }
+}
+
+#[macro_use]
+extern crate derive_more;
+
+use std::convert::TryInto;
+
+#[derive(Clone, Copy, TryInto)]
+enum MixedInts {
+    SmallInt(i32),
+    NamedBigInt { int: i64 },
+    TwoSmallInts(i32, i32),
+    NamedBigInts { x: i64, y: i64 },
+    Unsigned(u32),
+    NamedUnsigned { x: u32 },
+    Unit,
+}
+
+#[test]
+fn test_try_into() {
+    let i = MixedInts::SmallInt(42);
+    assert_eq!(TryInto::<i32>::try_into(i), Ok(42i32));
```

Comment:
Is this really the way this trait should be used? Can't you do something like:
```rust
assert_eq!(Ok(42i32), i.try_into())
```

Reply:
You can, but you can't do the same for the Err(()) cases.